cdap-app-fabric/src/main/java/co/cask/cdap/gateway/handlers/ProgramLifecycleHttpHandler.java

```diff
@@ -870,6 +877,16 @@ private ScheduleDetail readScheduleDetailBody(FullHttpRequest request, String sc
         "Schedule name in the body of the request (%s) does not match the schedule name in the path parameter (%s)",
         scheduleDetail.getName(), scheduleName));
     }
+
+    if (scheduleDetail.getProperties() != null) {
+      ProfileId profileId = SystemArguments.getProfileIdFromArgs(
+        new NamespaceId(scheduleDetail.getNamespace()), scheduleDetail.getProperties()).orElse(null);
+      if (profileId != null && profileStore.getProfileStatus(profileId) == ProfileStatus.DISABLED) {
```

Comment:
this seems like the wrong place to put this logic. I think it would make more sense in whatever class we have for storing schedule properties. Unless it's disconnected from this code path?

Reply:
fixed and moved to CoreSchedulerService